js/data/symbol_bucket.js

```diff
@@ -40,6 +40,8 @@ function SymbolBucket(options) {
 
     this.adjustedIconMaxSize = this.layer.getLayoutValue('icon-size', 18, zoomHistory);
     this.adjustedIconSize = this.layer.getLayoutValue('icon-size', this.zoom + 1, zoomHistory);
+
+    this.index = 0;
```

Comment:
My concern isn't using the index conceptually but adding additional state to the `SymbolBucket` class. It's ok for the context of this PR but makes my future a little harder.